### PR TITLE
delayOrRaise

### PR DESCRIPTION
**Pull request by i-walker**
Should I add an Example in the docs 
[noob Issue](https://github.com/arrow-kt/arrow/issues/1357)

### CODE REVIEW

File: modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/typeclasses/Async.kt [hunk 1]
```diff
@@ -22,227 +22,236 @@ typealias Proc<A> = ((Either<Throwable, A>) -> Unit) -> Unit
...
+     *   //sampleEnd
+     *   println(result)
+     * }
+     * ```
+     */
+    fun <A> defer(ctx: CoroutineContext, f: () -> Kind<F, A>): Kind<F, A> =
+            just(Unit).continueOn(ctx).flatMap { defer(f) }
+
+    /**
+     * Delay a computation on provided [CoroutineContext].
```
Comment (pakoito) [inline modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/typeclasses/Async.kt]: `Delay a branching computation on the provided [CoroutineContext].`

And you could use this opportunity to add a doc on the original `delayOrRaise` too as  `Delay a branching computation` :D

File: .bash_profile [hunk 2]
```diff
@@ -0,0 +1 @@
+export JAVA_HOME=$(/usr/libexec/java_home -v 1.8.0_201)
```
Comment (pakoito) [inline .bash_profile]: This shouldn't be on the repo hahaha

File: .gitignore [hunk 3]
```diff
@@ -130,3 +130,5 @@ reports/
 .DS_Store
 
 target/
+
+.bash_profile
```
Comment (pakoito) [inline .gitignore]: Shouldn't bash_profile be on your root anyway?

**Comment by pakoito** [comment]
Thank you for the PR!!

File: .gitignore (hunk 3 above)
Comment (i-walker) [reply .gitignore]: I tried putting it into the root, but the settings won't cascade to the actual arrow project. Can you help me, maybe I did it wrong. I have a Macbook Pro 10.14.3

Comment (pakoito) [reply .gitignore]: You may have needed to restart your terminal, or source the bash profile? I use zsh.

**Pull request merged**